File: cloud-run-function-kotlin/app/src/main/kotlin/org/phdezann/energy/service/EquivalenceConverter.kt
package org.phdezann.energy.service

import org.phdezann.energy.json.Response

class EquivalenceConverter {

    private val energyConverter = EnergyConverter()
    private val powerConverter = PowerConverter()
    private val jsonSerializer = JsonSerializer()

    fun convert(milliWattHour: Double): String {
        val energyMatches = energyConverter.convert(milliWattHour)
        val powerMatches = powerConverter.convert(milliWattHour)
        val response = Response(energyMatches, powerMatches)
        return jsonSerializer.writeValue(response)
    }
}

File: src/energy-converter/equivalence-converter.ts
import EnergyConverter, { EnergyMatch } from './energy-converter';
import PowerConverter, { PowerMatch } from './power-converter';

export type EquivalenceResult = {
  energy_matches: EnergyMatch[];
  power_matches: PowerMatch[];
};

class EquivalenceConverter {
  private energyConverter: EnergyConverter;
  private powerConverter: PowerConverter;

  constructor() {
    this.energyConverter = new EnergyConverter();
    this.powerConverter = new PowerConverter();
  }

  convert(milliWattHour: number): EquivalenceResult {
    const energyMatches = this.energyConverter.convert(milliWattHour);
    const powerMatches = this.powerConverter.convert(milliWattHour);
    return { energy_matches: energyMatches, power_matches: powerMatches };
  }
}

export default EquivalenceConverter;
